src/game/common/bitflags.h

```diff
@@ -242,77 +441,88 @@ template<int bits> void BitFlags<bits>::Parse(INI *ini, Utf8String *string)
         string->Clear();
     }
 
-    const char *token = ini->Get_Next_Token_Or_Null();
-    bool adjust = false;
     bool set = false;
+    bool adjust = false;
 
-    if (token != nullptr) {
-        while (strcasecmp(token, "NONE") != 0) {
-            // If we have a passed in Utf8String, add the tokens to it as we parse them.
-            if (string != nullptr) {
-                if (string->Is_Not_Empty()) {
-                    *string += " ";
-                }
+    for (const char *token = ini->Get_Next_Token_Or_Null(); token; token = ini->Get_Next_Token_Or_Null()) {
+        if (string) {
+            if (string->Is_Not_Empty()) {
+                string->Concat(" ");
+            }
+
+            string->Concat(token);
+        }
 
-                *string += token;
+        if (!strcasecmp(token, "NONE")) {
+            if (set || adjust) {
+                captainslog_error("you may not mix normal and +- ops in bitstring lists");
+                throw 0xDEAD0006;
             }
 
-            // If we have a plus or minus as first char, then that decides if bit is set or cleared.
-            // Otherwise just set.
-            if (*token == '+') {
-                captainslog_relassert(!set,
-                    0xDEAD0006,
-                    "File: '%s', Line: %d Trying to adjust when we already set.",
-                    ini->Get_Filename().Str(),
-                    ini->Get_Line_Number());
-
-                m_bits.set(INI::Scan_IndexList(token + 1, s_bitNamesList));
-                adjust = true;
-            } else if (*token == '-') {
-                captainslog_relassert(!set,
-                    0xDEAD0006,
-                    "File: '%s', Line: %d Trying to adjust when we already set.",
-                    ini->Get_Filename().Str(),
-                    ini->Get_Line_Number());
-
-                m_bits.reset(INI::Scan_IndexList(token + 1, s_bitNamesList));
-                adjust = true;
-            } else {
-                captainslog_relassert(!adjust,
-                    0xDEAD0006,
-                    "File: '%s', Line: %d Trying to set when we already adjusted.",
-                    ini->Get_Filename().Str(),
-                    ini->Get_Line_Number());
-
-                // If we haven't started setting yet, 0 it out.
-                if (!set) {
-                    m_bits.reset();
-                }
+            Clear();
+            return;
+        }
 
-                m_bits.set(INI::Scan_IndexList(token, s_bitNamesList));
-                set = true;
+        if (*token == '+') {
+            if (set) {
+                captainslog_error("you may not mix normal and +- ops in bitstring lists");
+                throw 0xDEAD0006;
             }
 
-            token = ini->Get_Next_Token_Or_Null();
+            Set(INI::Scan_IndexList(token + 1, s_bitNamesList), 1);
+            adjust = true;
+        } else if (*token == '+') {
```

Comment:
copy paste mistake?? shouldn't this be - or the one above it be -

Reply:
fixed.